prov/rxm/src/rxm_init.c

```diff
@@ -91,11 +95,12 @@ int rxm_info_to_core(uint32_t version, const struct fi_info *hints,
 	core_info->mode |= FI_RX_CQ_DATA | FI_CONTEXT;
 
 	if (hints) {
-		if (hints->caps & FI_TAGGED)
+		core_info->caps = hints->caps & RXM_PASSTHRU_CAPS;
+		if (hints->caps & (FI_ATOMIC | FI_TAGGED))
 			core_info->caps |= FI_MSG;
 
 		/* FI_RMA cap is needed for large message transfer protocol */
-		if (hints->caps & (FI_MSG | FI_TAGGED))
+		if (core_info->caps & FI_MSG)
 			core_info->caps |= FI_RMA;
```

Comment:
I think this may need to be more explicit.  E.g. FI_RMA | FI_READ | FI_WRITE | FI_REMOTE_WRITE | FI_REMOTE_READ.  Otherwise if FI_RMA | FI_WRITE is set by the app, the EP will not automatically get read or remote write access.

Reply:
Thanks, this makes sense; I've updated the code here and below.